File: client/database/tag.model.ts
import { Schema, model, Document, models, Model, Types } from 'mongoose'
import { IQuestion } from './question.model'

export interface ITag extends Document {
  name: string
  description: string
  questions: Types.ObjectId[] | IQuestion[]
  followers: Types.ObjectId[]
  createdOn: Date
}

const TagSchema = new Schema({
  name: { type: String, required: true, unique: true },
  description: { type: String, required: true },
  questions: [{ type: Schema.Types.ObjectId, ref: 'Question' }],
  followers: [{ type: Schema.Types.ObjectId, ref: 'User' }],
  createdOn: { type: Date, default: Date.now },
})

export const Tag: Model<ITag> = models.Tag || model<ITag>('Tag', TagSchema)

File: api/internal/data/Tag.model.go
package data

import "time"

type Tag struct {
	Id             int64     `json:"_id"`
	Name           string    `json:"name"`
	Description    string    `json:"description"`
	Questions      []int64   `json:"questions"`
	QuestionsCount int       `json:"questionsCount"`
	CreatedAt      time.Time `json:"createdAt"`
}
